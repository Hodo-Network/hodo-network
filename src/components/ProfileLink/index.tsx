import React, { FC } from "react";
import { Props } from "./types";

const ProfileLink: FC<Props> = ({ img, name }) => (
  <a href='#none' className='flex-shrink-0 w-full group block'>
    <div className='flex items-center'>
      <div>
        <img
          data-testid='image'
          className='inline-block h-9 w-9 rounded-full'
          src={img}
          alt={name}
        />
      </div>
      {/* <div className='ml-3'>
        <p
          className='text-sm leading-5 font-medium text-white'
          data-testid='name'>
          {name}
        </p>
        <p
          className='text-xs leading-4 font-medium text-gray-300 group-hover:text-gray-200 transition ease-in-out duration-150'
          data-testid='view-profile'>
          View profile
        </p>
      </div> */}
    </div>
  </a>
);

export default ProfileLink;
