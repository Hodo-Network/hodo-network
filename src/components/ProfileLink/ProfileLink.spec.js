import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ProfileLink from ".";

const img =
  "https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80";
const name = "Tom Cook";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ProfileLink />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<ProfileLink />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("contains profile image", () => {
  const { getByTestId } = render(<ProfileLink img={img} name={name} />);
  expect(getByTestId("image")).toHaveAttribute("src");
});

test.skip("contains user name", () => {
  const { getByTestId } = render(<ProfileLink img={img} name={name} />);
  expect(getByTestId("name")).toHaveTextContent("Tom Cook");
});

test.skip("contains view profile", () => {
  const { getByTestId } = render(<ProfileLink />);
  expect(getByTestId("view-profile")).toHaveTextContent("View profile");
});
