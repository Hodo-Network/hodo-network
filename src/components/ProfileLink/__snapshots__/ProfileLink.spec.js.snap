// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="flex-shrink-0 w-full group block"
  href="#none"
>
  <div
    className="flex items-center"
  >
    <div>
      <img
        className="inline-block h-9 w-9 rounded-full"
        data-testid="image"
      />
    </div>
  </div>
</a>
`;
